Sources/SectionScrubber.swift

```diff
@@ -18,55 +16,83 @@ public class SectionScrubber: UIView {
         case Visible
     }
 
-    private static let RightEdgeInset: CGFloat = 5.0
+    private let thumbMargin: CGFloat = -60
+
+    private let scrubberImageMargin: CGFloat = -6
 
-    /*
-     When calculating the NSIndexPath for the current scrubber position we need to use a x and a y coordinate,
-     the y coordinate is provided by how far you have scrubbed the scrubber, meanwhile we use a hardcoded x since
-     using 5 would ensure us that most of the time the first item in each row will be selected to retreive the
-     index path at certain location.
-    */
-    private static let initialXCoordinateToCalculateIndexPath = CGFloat(5)
+    public weak var delegate: SectionScrubberDelegate?
 
-    public var delegate: SectionScrubberDelegate?
+    public weak var dataSource: SectionScrubberDataSource?
+
+    private var adjustedContainerBoundsHeight: CGFloat {
+        guard let collectionView = self.collectionView else { return 0 }
+        return collectionView.bounds.height - (collectionView.contentInset.top + collectionView.contentInset.bottom + self.frame.height)
+    }
+
+    private var adjustedContainerOrigin: CGFloat {
+        guard let collectionView = self.collectionView else { return 0 }
+        guard let window = collectionView.window else { return 0 }
+
+        if collectionView.superview?.isKindOfClass(NSClassFromString("UICollectionViewControllerWrapperView")!) != nil {
```

Comment:
Because that is the superview only when contained inside a collection view controller. 👎